### PR TITLE
[fuzzydir] fix unicode character on Windows

### PR DESCRIPTION
**Pull request by dyphire**
Code taken from https://github.com/CogentRedTester/mpv-file-browser/blob/695362ee37c83492029dce2319aff7986a4cdf2a/addons/powershell.lua#L47-L65

### CODE REVIEW

File: fuzzydir.lua [hunk 1]
```diff
@@ -124,10 +124,17 @@ end
...
     if powershell_version >= 3 then
         return call_command({
-            "powershell.exe",
+            "powershell",
             "-NoProfile",
             "-Command",
-            "& { Get-ChildItem -LiteralPath FileSystem::\"" .. path .. "\" -Directory | foreach { $_.Name } }",
+            [[
+            $dirs = Get-ChildItem -LiteralPath ]] .. string.format("%q", path) .. [[ -Directory
+            foreach ($n in $dirs.Name) {
```
Comment (sibwaf) [inline fuzzydir.lua]: Not sure if this syntax works to be honest - at least doesn't seem to with PS 2 I have in a VM. Though, this optimization doesn't work in general with PS 2 as it lacks the `-Directory` flag for `Get-ChildItem`, so maybe it's fine with newer versions?..

In any case, could you please change it to something like this just to be safe?
```
foreach($dir in $dirs) {
    $u8clip = [System.Text.Encoding]::UTF8.GetBytes($dir.Name)
    ...
}
```

We also don't really need the trailing slash, it's purely aesthetical.

File: fuzzydir.lua [hunk 2]
```diff
@@ -124,10 +124,17 @@ end
 function fast_readdir(path)
     if powershell_version >= 3 then
         return call_command({
-            "powershell.exe",
+            "powershell",
             "-NoProfile",
             "-Command",
-            "& { Get-ChildItem -LiteralPath FileSystem::\"" .. path .. "\" -Directory | foreach { $_.Name } }",
+            [[
+            $dirs = Get-ChildItem -LiteralPath ]] .. string.format("%q", path) .. [[ -Directory
```
Comment (sibwaf) [inline fuzzydir.lua:131]: Missing the `FileSystem::` part right before the path, will break with direct access to network shares (paths like `\\server\name\directory`) if I remember correctly the results from my previous tests.

The resulting string should look like `FileSystem::"path"`

Comment (sibwaf) [reply fuzzydir.lua:131]: Or not. Retested this, seems to be working fine for some reason.

File: fuzzydir.lua (hunk 1 above)
Comment (dyphire) [reply fuzzydir.lua]: > Not sure if this syntax works to be honest - at least doesn't seem to with PS 2 I have in a VM. Though, this optimization doesn't work in general with PS 2 as it lacks the `-Directory` flag for `Get-ChildItem`, so maybe it's fine with newer versions?..

In my tests on Win10 and Win11, it worked fine, and earlier systems had no conditional testing.

> In any case, could you please change it to something like this just to be safe?
> 
> ```
> foreach($dir in $dirs) {
>     $u8clip = [System.Text.Encoding]::UTF8.GetBytes($dir.Name)
>     ...
> }
> ```
> 
> We also don't really need the trailing slash, it's purely aesthetical.

Done

**Comment by sibwaf** [comment]
LGTM, will merge after feedback from @AdventurerRussia. Thanks!

**Comment by AdventurerRussia** [comment]
yes, everything works fine, thanks for the fixes

**Pull request merged**